### PR TITLE
Added missing idle args in XPendingExtArgs

### PR DESCRIPTION
**Pull request by parvez0**
Made changes to XPendingExt method for allowing the usage of idle argument if defined in XPendingExtArgs

### CODE REVIEW

**Comment by monkey92t** [comment]
@parvez0 

Hi~ this modification is wrong, you need to check the order of the parameters, redis-server does not handle it rigorously.
https://github.com/redis/redis/blob/6.2.1/src/t_stream.c#L2572-L2629

In addition, we need to complete it in the unit test

**Comment by parvez0** [comment]
What is the command for running test and also which platform is it supported by make test. If you can point me to a developer guide that will be great. As for those changes I'll do it thanks for the clarification

**Comment by monkey92t** [comment]
You can follow github workflow, link: https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions

**Comment by monkey92t** [comment]
According to the redis documentation, the unit of idle is milliseconds, not nanoseconds

**Comment by parvez0** [comment]
I have looked at the MinRetryBackoff and other fields there is no conversion which is happening for time. Duration if specified I thought the developer has to take care of that setting so should I change it so that at the time of adding it should convert it to millisec

**Comment by parvez0** [comment]
@monkey92t can you verify the changes now

File: commands_test.go [hunk 1]
```diff
@@ -4229,6 +4229,7 @@ var _ = Describe("Commands", func() {
 				infoExt, err := client.XPendingExt(ctx, &redis.XPendingExtArgs{
 					Stream:   "stream",
 					Group:    "group",
+					Idle:     time.Duration(0),
```
[the remark this answers is not in the document]
Comment (parvez0) [reply commands_test.go:4232]: Sure I'll write a separate test case for Idle time

File: commands.go [hunk 2]
```diff
@@ -1819,7 +1820,11 @@ type XPendingExtArgs struct {
 
 func (c cmdable) XPendingExt(ctx context.Context, a *XPendingExtArgs) *XPendingExtCmd {
 	args := make([]interface{}, 0, 7)
-	args = append(args, "xpending", a.Stream, a.Group, a.Start, a.End, a.Count)
+	args = append(args, "xpending", a.Stream, a.Group)
+	if a.Idle != 0 {
+		args = append(args, "idle", a.Idle * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (parvez0) [reply commands.go:1825]: formatMs should be a new function right where should it go I mean in which file

[the remark this answers is not in the document]
Comment (monkey92t) [reply commands.go:1825]: https://github.com/go-redis/redis/blob/7d319aeb5735d0f2fef4e513bb65c7f1ce9e38a8/commands.go#L22-L32

File: commands_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (parvez0) [reply commands_test.go:4232]: @monkey92t  sorry for bugging you, I have written the test case but in order to verify it I have to add a simple time.Sleep in between of the test cases and I am not sure if that is the right way to do it. I just wanted to confirm once

[the remark this answers is not in the document]
Comment (parvez0) [reply commands_test.go:4232]: // sleeping for 2 millisecond to test for idle time
				time.Sleep(5 * time.Millisecond)
				infoExt, err = client.XPendingExt(ctx, &redis.XPendingExtArgs{
                                         // other fields
					Idle:     2 * time.Millisecond,
				}).Result()

[the remark this answers is not in the document]
Comment (parvez0) [reply commands_test.go:4232]: and also there are some linting errors from my previous commit which is breaking the checks on mine can you tell what should I do about that

[the remark this answers is not in the document]
Comment (monkey92t) [reply commands_test.go]: We only need to set a large enough time to prove the normal execution of the idle parameter.
500 milliseconds is too short, it is easy to make mistakes

```go
args.Idle = 72 * time.Hour
infoExt, err = client.XPendingExt(ctx, args).Result()
Expect(err).NotTo(HaveOccurred())
Expect(infoExt).To(HaveLen(0))
```

**Comment by monkey92t** [comment]
Close #1746

**Comment by monkey92t** [comment]
Fixed #1749

**Comment by jalmeter** [comment]
@monkey92t - Do you have a guess for when this will be released?

**Comment by vmihailenco** [comment]
I've pushed v8.8.3

**Pull request merged**